### PR TITLE
IdP error events

### PR DESCRIPTION
**Pull request by martinthomson**
There's currently no way to learn of problems with the IdP processes.  This adds two error events that cover generation and validation separately.

This depends on #11.  It's based on by feedback in http://lists.w3.org/Archives/Public/public-webrtc/2013Dec/0104.html

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -1634,29 +1646,39 @@
...
+                <p>A stream could have contents that are inaccessible to the
+                application.  This can be due to being marked with a
+                <var>peerIdentity</var> option or anything that would make a
+                stream <a
+                href="http://www.w3.org/html/wg/drafts/html/master/infrastructure.html#cors-cross-origin">CORS
+                cross-origin</a>.  These streams can be added to the <a
+                href="#local-streams-set">local streams set</a> but content MUST
+                NOT be transmitted, with one exception.</p>
+
+                <p>A stream marked with the <var>peerIdentity</var> option can
```
Comment (stefhak) [inline webrtc.html:1669]: In this section, is there not an ambiguity between a "stream" (=MediaStream I presume) and track (i.e. MediaStreamTrack)? 

And in principle I think you could end up with tracks (in the same stream) having different identites.

(I continue to think things could be much simpler if the PC dealt with tracks only)

File: webrtc.html [hunk 2]
```diff
@@ -1634,29 +1646,39 @@
...
+                href="sec.identity-verify-assertion">validated the identity</a>
+                of the peer to have the same identity as the value of the
+                <var>peerIdentity</var> option on the stream.</p>
+
+                <p>All other streams that are not accessible to the application
+                MUST NOT be sent to the peer, with silence (audio), black frames
+                (video) or equivalently absent content being sent in place of
+                stream content.</p>
+
+                <p>Note that this property can change over time.</p>
```
Comment (stefhak) [inline webrtc.html:1680]: Unclear what is meant (what can change?)

File: webrtc.html [hunk 3]
```diff
@@ -4143,149 +4163,187 @@ <h4 id="sec.create-identity-proxy">Instantiating an IdP Proxy</h4>
...
+      cause <code><a href="#dom-peerconnection-setremotedescription">setRemoteDescription</a></code>
+      to fail.</p>
+
+      <p>The browser SHOULD limit the time that it will allow for this process.
+      This includes both the loading of
+      the <a href="#sec.identity-proxy-communications">IdP proxy</a> and the
+      identity assertion validation.  Failure to do so potentially causes the
+      corresponding operation to take an indefinite amount of time.  This timer
+      can be cancelled when the IdP produces a response.  The timer running to
+      completion can be treated as equivalent to an error from the IdP.</p>
```
Comment (stefhak) [inline webrtc.html:4337]: I think the exact same section is inserted earlier. Mistake, or meant to be so?

Comment (martinthomson) [inline webrtc.html:4337]: There is text for assertion generation AND validation.  Yes, the text is duplicated.

File: webrtc.html (hunk 2 above)
Comment (martinthomson) [inline webrtc.html:1680]: I might have to remove this, it's a little tricky to explain and not relevant until we get to explaining how remote streams behave.

Streams originating from RTCPeerConnection have to change their properties.  Prior to knowing for certain if a stream should be isolated, we have to isolate the stream.

Streams that are produced from <video> can change isolation properties too, but that's a completely separate problem.

File: webrtc.html (hunk 1 above)
Comment (martinthomson) [inline webrtc.html:1669]: I went back and forth on this.  I found it easier to talk about the attribute being a stream attribute.

**Pull request merged**